server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -75,6 +80,10 @@ private static FieldFetcher create(
 
             for (String field : context.getMatchingFieldNames(fieldPattern)) {
                 MappedFieldType ft = context.getFieldType(field);
+                if (ft instanceof LookupRuntimeFieldType lookupField) {
```

Comment:
I don't have a concrete suggestion yet, but it'd be really nice if we could structure this to avoid type checking. We've worked really hard to keep the mappers modular, so there's no "special casing" in core search code for specific types. In fact many mappers live in modules, so it's not even possible to check their types within search.

Reply:
Same here. I thought on this before decided to use the type checking. I will play a bit more if I can find a better option than the type checking.